src/OpenSage.Game/Logic/Object/Helpers/TempWeaponBonusHelper.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace OpenSage.Logic.Object.Helpers
+{
+    internal sealed class TempWeaponBonusHelper : ObjectHelperModule
+    {
+        private int _unknownInt;
+
+        internal override void Load(StatePersister reader)
+        {
+            reader.PersistVersion(1);
+
+            base.Load(reader);
+
+            reader.PersistInt32(ref _unknownInt);
+            if (_unknownInt != -1)
```

Comment:
I had `if (_unknownInt != 0 && _unknownInt != -1)`, so 0 may have occurred for some reason in one of my saves.

Reply:
Interesting. I'll leave it like it is for now but that's good to know. I'll try to upload some test .sav files to give us some regression tests for this code.